File: go/problems/9_palindrome_number_test.go
package main

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestIsPalindrome(t *testing.T) {
	tests := []struct {
		x    int
		want bool
	}{
		{
			x:    121,
			want: true,
		},
		{
			x:    -121,
			want: false,
		},
		{
			x:    10,
			want: false,
		},
		{
			x:    0,
			want: true,
		},
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("is palindrome for %d should return %t", tt.x, tt.want), func(t *testing.T) {
			got := isPalindrome(tt.x)
			assert.Equal(t, tt.want, got, "got %t want %t", got, tt.want)
		})
	}
}

func TestIsPalindromeStr(t *testing.T) {
	tests := []struct {
		x    int
		want bool
	}{
		{
			x:    121,
			want: true,
		},
		{
			x:    -121,
			want: false,
		},
		{
			x:    10,
			want: false,
		},
		{
			x:    0,
			want: true,
		},
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("is palindrome for %d should return %t", tt.x, tt.want), func(t *testing.T) {
			got := isPalindromeStr(tt.x)
			assert.Equal(t, tt.want, got, "got %t want %t", got, tt.want)
		})
	}
}

File: ts/problems/9-palindrome-number.test.ts
import { describe, expect, test } from "vitest";
import { isPalindrome, isPalindromeStr } from "./9-palindrome-number";

describe("isPalindrome", () => {
  test.each([
    {
      x: 121,
      want: true,
    },
    {
      x: -121,
      want: false,
    },
    {
      x: 10,
      want: false,
    },
    {
      x: 0,
      want: true,
    },
  ])("check if $x is a palindrome and expects $want", ({ x, want }) => {
    expect(isPalindrome(x)).toBe(want);
  });
});

describe("isPalindromeStr", () => {
  test.each([
    {
      x: 121,
      want: true,
    },
    {
      x: -121,
      want: false,
    },
    {
      x: 10,
      want: false,
    },
    {
      x: 0,
      want: true,
    },
  ])("check if $x is a palindrome and expects $want", ({ x, want }) => {
    expect(isPalindromeStr(x)).toBe(want);
  });
});
